### PR TITLE
Fix - Tag Bugs

### LINKED ISSUE #282: [PE-D] DeleteTag feature does not remove tags from all contacts
Inputting the command 'deletetag INDEX' removes the tag from almost all contacts except one.

In the screenshot below, the tag 'friends' was deleted, but contact number 2, 'Bernice Yuu' still has the 'friends' tag attached to her.
 
deletetag

 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: reignnz/ped#1

### LINKED ISSUE #284: [PE-D] FindByTag feature includes overlapping tags
`findbytag /tag` command will include overlapped tags
for instance, I input `findbytag /tag` but received 3 contacts, one of which included a tag which was `tagg`, and not `tag`
By right, i should have received 2 contacts

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: zxgoh/ped#3

### LINKED ISSUE #294: [PE-D] Tag overflow
Extremely long tags overflow beyond the Ui of the address book, which means that users cannot see the full tag

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: iztanpy/ped#5

### LINKED ISSUE #297: [PE-D] Related to Issue#1, after deleting a tag, finding the same tag still works
In the screenshot below, we can see that Tags does not have the tag 'friends'. However, when search for the deleted tag 'friends', by inputting 'findbytag t/friends', it lists one contact. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: reignnz/ped#3

### PR DESCRIPTION
**Pull request by KwanHW**
# Changes
This PR addresses the bugs found in PE-D relating to Tags.

## Tag overflow fix
image

# Related Issues
- Closes #282 
- Closes #284 
- Closes #294 
- Closes #297

### CODE REVIEW

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm

File: src/main/java/seedu/contax/logic/commands/FindByTagCommand.java [hunk 1]
```diff
@@ -29,6 +29,13 @@ public FindByTagCommand(TagNameContainsKeywordsPredicate predicate) {
     public CommandResult execute(Model model) {
         requireNonNull(model);
         model.updateFilteredPersonList(predicate);
+
+        // If tag does not exist in the list, just show that there are no persons
+        if (!predicate.existsInTagList(model.getFilteredTagList())) {
+            return new CommandResult(String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, 0),
+                    GuiListContentType.PERSON);
+        }
```
Comment (sharpstorm) [inline src/main/java/seedu/contax/logic/commands/FindByTagCommand.java:37]: This is redundant, if you really want to do a defensive mechanism, add a always false predicate in here.

**Review by sharpstorm — APPROVED** [review]
LGTM.

File: src/main/java/seedu/contax/logic/commands/FindByTagCommand.java (hunk 1 above)
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/commands/FindByTagCommand.java]: Fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/324/commits/b6d8de73f1cb326c8b411ffc056e66df4c0d51cf

**Comment by KwanHW** [comment]
Will need another review as commits that fix #282 was lost.

**Review by sharpstorm — CHANGES_REQUESTED** [review]
You accidentally touched the onboarding FXML and prompt class.

**Review by sharpstorm — APPROVED** [review]
Sorry about the request for change, it looks like it was only a indent fix.

**Pull request merged**